### PR TITLE
[docs] Disaster recovery etcd doc

### PR DESCRIPTION
**Pull request by name212**
## Description
Add disaster recovery etcd doc.

## Why do we need it, and what problem does it solve?
It helps recovery etcd cluster. 

## Changelog entries

```changes
section: docs
type: feature
summary: Disaster recovery etcd documentation.
impact_level: low
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'deploy/web/stage'...

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+	/go/src/go.etcd.io/etcd/release/etcd/server/etcdmain/main.go:40 +0x13f
+main.main()
+	/go/src/go.etcd.io/etcd/release/etcd/server/main.go:32 +0x45
+```
+Исходя из [issue](https://github.com/etcd-io/etcd/issues/11949) такая ошибка может появиться и после некорректного завершения работы etcd.
+
+#### Решение 1
+[Источник](https://github.com/etcd-io/etcd/issues/11949#issuecomment-1029906679)
+- останавливаем etcd (перемещаем манифест пода)
+- беками файлы
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:53]:
```suggestion
- бекапим файлы
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+```bash
+kubectl -n kube-system exec -ti ETCD_POD_ON_AFFECTED_HOST -- /bin/sh -c 'ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ defrag --command-timeout=60s'
+```
+
+#### Решение 2
+Если решение 1 не помогло, то:
+- загружаем на сервер [etcdctl](https://github.com/etcd-io/etcd/releases), желательно той же версии, что и на сервере.
+- останавливаем etcd (перемещаем манифест пода)
+- увеличиваем `quota-backend-bytes` в манифесте, если это необходимо
+- беками файлы, например, в `/var/lib/etcd-backup`
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:75]:
```suggestion
- бекапим файлы, например, в `/var/lib/etcd-backup`
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+```bash
+ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ alarm disarm
+```
+- дефрагментируем etcd если это необходимо:
+```bash
+ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ defrag --command-timeout=60s
+```
+
+### Восстанавливаемся из бекапа
+- загружаем на сервер [etcdctl](https://github.com/etcd-io/etcd/releases), желательно той же версии, что и сервере
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:93]:
```suggestion
- загружаем на сервер [etcdctl](https://github.com/etcd-io/etcd/releases), желательно той же версии, что и на сервере
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+## Multi-master
+
+### Потеря кворума
+- добавляем в манифест аргумент `--force-new-cluster` на оставшейся ноде
+- ждем поднятия etcd
+- удаляем аргумент `--force-new-cluster`
+Далее, если ноды были потеряны безвозвратно, то добавлям новые через `dhctl converge` или вручную, если кластер статический.
+
+Если ноды возвратились, то они уже не являются членами кластера.
+- на нодах, которые не являются членами кластера выполняем следующие операции:
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:108]:
```suggestion
На узлах, которые не являются членами кластера выполняем следующие операции:
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+- останавливаем etcd (перемещаем манифест пода)
+- беками файлы (на всякий случай)
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:110]:
```suggestion
- бекапим файлы (на всякий случай)
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+- удаляем папку c данными `rm -rf /var/lib/etcd/`
+
+Возвращаемся на рабочий кластер, и перезапускаем демонсет `d8-control-plane-manager`:
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:113]:
```suggestion
Возвращаемся на рабочий кластер, и перезапускаем DaemonsSet `d8-control-plane-manager`:
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+```bash
+kubectl -n kube-system rollout restart daemonset d8-control-plane-manager
+```
+Ожидаем переката всех control-plane подов и переход их в состояние `Ready` и смотрим что все etcd инстансы стали участниками кластера.
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:117]:
```suggestion
Ожидаем переката всех Pod'ов control-plane и переход их в состояние `Ready` и смотрим что все etcd инстансы стали участниками кластера.
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+```
+ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ member list -w table
+```
+### Полная потеря данных и восстановление из бекапа
+
+На всех нодах:
+- останавливаем etcd (перемещаем манифест пода)
+- беками файлы (на всякий случай)
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:125]:
```suggestion
- выполняем резервное копирование файлов (на всякий случай)
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+- удаляем папку c данными `rm -rf /var/lib/etcd/`
+- восстанавливаем на одной ноде как для [single-мастера](#восстанавливаемся-из-бекапа)
+- перезапускаем демонсет `d8-control-plane-manager`:
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:128]:
```suggestion
- перезапускаем DaemonSet `d8-control-plane-manager`:
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+```bash
+kubectl -n kube-system rollout restart daemonset d8-control-plane-manager
+```
+Ожидаем переката всех control-plane подов и переход их в состояние `Ready` и смотрим что все etcd инстансы стали участниками кластера:
```
Comment (z9r5) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:132]:
```suggestion
Ожидаем переката всех Pod'ов control-plane и переход их в состояние `Ready` и смотрим что все etcd-инстансы стали участниками кластера:
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 11]
```diff
@@ -0,0 +1,135 @@
...
+```bash
+kubectl -n kube-system exec -ti ETCD_POD_ON_AFFECTED_HOST -- /bin/sh -c 'ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ defrag --command-timeout=60s'
+```
+
+#### Решение 2
+Если решение 1 не помогло, то:
+- загружаем на сервер [etcdctl](https://github.com/etcd-io/etcd/releases), желательно той же версии, что и на сервере.
+- останавливаем etcd (перемещаем манифест пода)
+- увеличиваем `quota-backend-bytes` в манифесте, если это необходимо
+- бекапим файлы, например, в `/var/lib/etcd-backup`
```
Comment (konstantin-axenov) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:75]:
```suggestion
- бекапим файлы, копируя папку `cp -r /var/lib/etcd /var/lib/etcd-backup`
```

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+- удаляем папку c данными `rm -rf /var/lib/etcd/`
+- восстанавливаем базу `ETCDCTL_API=3 etcdctl snapshot restore /var/lib/etcd-backup/member/snap/db --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/  --data-dir=/var/lib/etcd --skip-hash-check`.
+- пытаемся запустить etcd
```
Comment (konstantin-axenov) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:78]: How to do it?

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 13]
```diff
@@ -0,0 +1,135 @@
...
+```
+ETCDCTL_API=3 etcdctl --cacert /etc/kubernetes/pki/etcd/ca.crt --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ member list -w table
+```
+### Полная потеря данных и восстановление из бекапа
+
+На всех нодах:
+- останавливаем etcd (перемещаем манифест пода)
+- выполняем резервное копирование файлов (на всякий случай)
+- удаляем папку c данными `rm -rf /var/lib/etcd/`
+- восстанавливаем на одной ноде как для [single-мастера](#восстанавливаемся-из-бекапа)
```
Comment (konstantin-axenov) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:127]: Don't we need to make single-master first if we have multi-master?

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 14]
```diff
@@ -0,0 +1,135 @@
+# Восстановление работоспособности ETCD
```
Comment (konstantin-axenov) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:1]: Rename file to ETCD_RECOVERY_RU.md

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+```
+
+Старт etcd после попытки восстановления:
+```bash
+mv ~/etcd.yaml /etc/kubernetes/manifests/etcd.yaml
+```
+
+## Single-master
+
+### Ситуация 1
```
Comment (konstantin-axenov) [inline modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md:23]: Strange naming) I can't see number 2

Comment (name212) [reply modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md]: fixed :-)

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md (hunk 14 above)
Comment (name212) [reply modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md]: done

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md (hunk 13 above)
Comment (name212) [reply modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md]: yep, i added it

File: modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md (hunk 12 above)
Comment (name212) [reply modules/040-node-manager/docs/internal/ETC_RECOVERY_RU.md]: fixed

**Comment by name212** [comment]
@z9r5 I think we send this doc to translate

File: modules/040-node-manager/docs/internal/ETCD_RECOVERY_RU.md [hunk 16]
```diff
@@ -0,0 +1,279 @@
...
+    --cert /etc/kubernetes/pki/etcd/ca.crt --key /etc/kubernetes/pki/etcd/ca.key --endpoints https://127.0.0.1:2379/ defrag --command-timeout=60s'
+  ```
+
+#### Решение 2
+
+Следующим вариантом решения можно воспользоваться, если первый вариант решения не помог.
+
+Выполните следующие шаги:
+
+- загрузите на сервер [etcdctl](https://github.com/etcd-io/etcd/releases) (желательно той же версии, что и на версия etc на сервере);
```
Comment (kirkonru) [inline modules/040-node-manager/docs/internal/ETCD_RECOVERY_RU.md:96]:
```suggestion
- загрузите [etcdctl](https://github.com/etcd-io/etcd/releases) на сервер (желательно той же версии, что и версия etcd на сервере);
```

**Review by kirkonru — COMMENTED** [review]
Please, take a look at my suggestions. Minor fixes + rephrasing (I think the phrases sound better that way).

**Pull request merged**